server/pulp/server/db/manage.py

```diff
@@ -165,7 +165,16 @@ def ensure_database_indexes():
 
     # Load all the model classes that the server knows about and ensure their indexes as well
     plugin_manager = PluginManager()
-    for model_class in plugin_manager.unit_models.itervalues():
+    for unit_type, model_class in plugin_manager.unit_models.items():
+        unit_key_index = {'fields': model_class.unit_key_fields, 'unique': True}
+        for index in model_class._meta['indexes']:
+            if unit_key_index == index:
+                raise Exception(_("Content unit type '%s' explicitly defines an index for its unit"
```

Comment:
I know that I suggested adding this, but I think I want a second opinion on whether or not this is even necessary. Right now I could take it or leave it.

Maybe the right solution is adding information about this to the docs for adding a new type, but I believe those are all so out of date that adding it here would be a ton of work for you.